compiler/src/test/scala/org/scalajs/core/compiler/test/OptimizationTest.scala

```diff
@@ -49,6 +49,46 @@ class OptimizationTest extends JSASTTest {
     }
   }
 
+  @Test
+  def testArrayApplyOptimization: Unit = {
+    /* Make sure Array(...) is optimized away completely for several kinds
+     * of data types, with both the generic overload and the ones specialized
+     * for primitives.
+     */
+    """
+    class A {
+      val a = Array(5, 7, 9, -3)
+      val b = Array("hello", "world")
+      val c = Array('a', 'b')
+      val d = Array(Nil)
+      val e = Array(5.toByte, 7.toByte, 9.toByte, -3.toByte)
+    }
+    """.
+    hasNot("any of the Array.apply methods") {
+      case js.Apply(js.LoadModule(jstpe.ClassType("s_Array$")), js.Ident(methodName, _), _)
+          if methodName.startsWith("apply__") =>
```

Comment:
Could we make this more restrictive? IIUC the goal of this is to not require scalalib for this. So maybe we should simply check `js.LoadModule(jstpe.ClassType("s_Array$"))` is not there? (or maybe even *any* `js.LoadModule` tree)?

Reply:
Done, updated. Just any `LoadModule` didn't work, but since I'm on mobile I couldn't figure out why.